Buildings/Applications/DHC/EnergyTransferStations/CoolingIndirect.mo

```diff
@@ -0,0 +1,287 @@
+    final k=k,
+    Td=Td,
+    final yMax=yMax,
+    final yMin=yMin,
+    final Ti=Ti,
+    wp=wp,
+    wd=wd,
+    Ni=Ni,
+    Nd=Nd,
+    final initType=Modelica.Blocks.Types.InitPID.InitialOutput,
+    xi_start=xi_start,
+    xd_start=xd_start,
+    final y_start=0,
+    final reverseAction=reverseAction) "Controller"
+    annotation (Placement(transformation(extent={{-90,-10},{-70,10}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisSup(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m1_flow_nominal)
+    "District-side (primary) supply temperature sensor"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={-80,60})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisRet(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m1_flow_nominal)
+    "District-side (primary) return temperature sensor"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={80,60})));
+
+  Modelica.Blocks.Sources.RealExpression powCal(
+    y=senMasFlo.m_flow*cp*(senTDisRet.T - senTDisSup.T))
+    "Calculated power demand"
+    annotation (Placement(transformation(extent={{-100,-100},{-20,-80}})));
+
+  Modelica.Blocks.Interfaces.RealOutput Q_flow(
+    quantity="Power",
+    unit="W",
+    displayUnit="kW")
+    "Measured power demand at the ETS"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=-90,
+        origin={90,-130}), iconTransformation(
+        extent={{-10,-10},{10,10}},
+        origin={70,-110},
+        rotation=-90)));
+
+  Modelica.Blocks.Interfaces.RealOutput Q(
+    quantity="Energy",
+    unit="J",
+    displayUnit="kWh")
+    "Measured energy consumption at the ETS"
+    annotation (Placement(
+        transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=-90,
+        origin={50,-130}), iconTransformation(
+        extent={{-10,-10},{10,10}},
+        origin={30,-110},
+        rotation=-90)));
+
+  Modelica.Blocks.Continuous.Integrator int(k=1) "Integration"
+    annotation (Placement(transformation(extent={{0,-120},{20,-100}})));
+
+  Buildings.Fluid.Sensors.MassFlowRate senMasFlo(redeclare package Medium = Medium)
+    annotation (Placement(transformation(extent={{-60,50},{-40,70}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort TBuiRet(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m2_flow_nominal)
+    "Building-side (secondary) return temperature"
+    annotation (Placement(transformation(extent={{-70,-70},{-90,-50}})));
+  Buildings.Fluid.Actuators.Valves.TwoWayQuickOpening val(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m1_flow_nominal,
+    dpValve_nominal=dpValve_nominal,
+    riseTime(displayUnit="s") = 60,
+    y_start=0) "District-side (primary) control valve"
+    annotation (Placement(transformation(extent={{-30,70},{-10,50}})));
```

Comment:
Ideally, keep the icon the same as when you drag and place it here, don't flip it.

Reply:
Thanks for noting this. All rotations have been removed from icons in this model for the next commit.